lending_pool/src/lib.rs

```diff
@@ -28,19 +29,17 @@ pub trait LendingPool:
     #[endpoint(deposit)]
     fn deposit_endpoint(
         &self,
-        #[var_args] caller: OptionalArg<Address>,
         #[payment_token] asset: TokenIdentifier,
         #[payment_amount] amount: Self::BigUint,
+        #[var_args] caller: OptionalArg<Address>,
     ) -> SCResult<()> {
-        let initial_caller = caller
-            .into_option()
-            .unwrap_or_else(|| self.blockchain().get_caller());
+        let initial_caller = self.caller_from_option_or_sender(caller);
 
-        require!(amount > 0, "amount must be greater than 0");
-        require!(!initial_caller.is_zero(), "invalid address provided");
+        self.require_amount_greater_than_zero(&amount)?;
+        self.require_non_zero_address(&initial_caller)?;
 
         let pool_address = self.get_pool_address(&asset);
```

Comment:
Could use the already existing function `get_pool_address_non_zero`.

Reply:
done